pallene/typedecl.lua

```diff
@@ -82,4 +82,35 @@ function typedecl.declare(module, mod_name, type_name, constructors)
     end
 end
 
+-- Returns true if the 2nd argument is a prefix of the first.
+--
+-- @param tag: The type name (string)
+-- @param tag_prefix: The prefix to test (string)
+
+function typedecl.tag_matches(tag, tag_prefix)
+    return type(tag) == "string"
+        and string.find(tag, tag_prefix, 1, true) ~= nil
```

Comment:
Oh shoot, I forgot about that too.

 Yeah, I definitely have intentions of making contributions in the future since I love compilers and have been exploring type systems as of late.

I actually messed this up because I was coding on my Linux PC, If I manage to get Pallene built up and running on windows these silly errors most likely won't persist.